Core/SaveState.cpp

```diff
@@ -471,6 +460,63 @@ namespace SaveState
 		}
 	}
 
+	void LoadSlot(const Path &gameFilename, int slot, Callback callback, void *cbUserData)
+	{
+		Path fn = GenerateSaveSlotFilename(gameFilename, slot, STATE_EXTENSION);
+		if (!fn.empty()) {
+			// This add only 1 extra state, should we just always enable it?
+			if (g_Config.bEnableStateUndo) {
+				Path backup = GetSysDirectory(DIRECTORY_SAVESTATE) / LOAD_UNDO_NAME;
+				
+				auto saveCallback = [=](Status status, const std::string &message, void *data) {
+					if (status != Status::FAILURE) {
+						DeleteIfExists(backup);
+						File::Rename(backup.WithExtraExtension(".tmp"), backup);
+						g_Config.sStateLoadUndoGame = GenerateFullDiskId(gameFilename);
+						Load(fn, slot, callback, cbUserData);
+					} else if (callback) {
+						callback(status, message, data);
+					}
+				};
+
+				if (!backup.empty()) {
+					Save(backup.WithExtraExtension(".tmp"), -2, saveCallback, cbUserData);
+				} else {
+					auto sy = GetI18NCategory("System");
+					if (callback)
+						callback(Status::FAILURE, sy->T("Failed to save state for load undo. Error in the file system."), cbUserData);
```

Comment:
Hm, if I'm trying to load a state and saving fails, maybe it's my disk being full... I feel like I'd still want to load state in this situation, and be frustrated if the emulator wouldn't let me load just because I can't save.

-[Unknown]